TESTS/mbed_drivers/ticker/main.cpp

```diff
@@ -104,6 +104,10 @@ void increment_multi_counter(void)
  */
 void test_case_1x_ticker()
 {
+#if defined(__ARM_FM)
```

Comment:
If it's required to skip tests, the pattern that we use is to add the #if def in the `Cases cases[]` array.
This makes determining if tests are being skipped much easier to follow.

Please make this change to this and the other tests.

Reply:
Actually, that was my first attempt. with more then one case it just fine.
However,  some of the tests only have one test case, e.g. [here](https://github.com/ARMmbed/mbed-os/blob/891c4ed244407dfee0f4ceb76d305ccc0628ad68/TESTS/mbedmicro-rtos-mbed/basic/main.cpp#L111)
I can't skip the only test case in the `Case case[ ]` . any suggestions on this case ? should I block the whole test file as `NOT_SUPPORTTED`

I took the currently method purely becasue it give a message while skipping the tests. I admit it not as easy as the other on to follow in the code level.

Reply:
Ah, that's a good point.

As an example for how things are done right now (doesn't mean that they should be like this, just means that this is how we do it right now), take a look at this test: https://github.com/ARMmbed/mbed-os/blob/master/TESTS/mbed_hal/common_tickers/main.cpp

There's an `ifdef` on the top of the file for targets that don't have the supported feature, and another `ifdef` for if a different feature is available. Please follow this pattern for this PR, and I would appreciate an issue being opened for this alternate method.